src/components/v5/common/Fields/InputBase/FormInputBase.tsx

```diff
@@ -47,7 +48,14 @@ const FormInputBase: FC<FormInputBaseProps> = ({
         if (type === 'number') {
           onChange(Number.isNaN(valueAsNumber) ? 0 : valueAsNumber);
         } else {
-          onChange(inputValue);
+          onChange(
+            shouldAllowNumbers
+              ? inputValue
+                  .replace(/[^0-9.]/g, '')
+                  .replace(/(\..*?)\..*/g, '$1')
+                  .replace(/^0[^.]/, '0')
+              : inputValue,
```

Comment:
Rather than allowing numbers, isn't this disallowing non-numbers? Setting `shouldAllowNumbers` to false still allows them

Reply:
Renamed to `shouldAllowOnlyNumbers`